handlers_test.go

```diff
@@ -176,8 +177,43 @@ func TestHashHandler(t *testing.T) {
 	}
 }
 
+func TestReadiness(t *testing.T) {
+	cfg := defaultCfg
+	cfg.WaitForApp = false
+	e := createEnclave(&cfg)
+	if err := e.Start(); err != nil {
+		t.Fatal(err)
+	}
+	defer e.Stop() //nolint:errcheck
+
+	nitridingSrv := fmt.Sprintf("https://127.0.0.1:%d", e.cfg.ExtPort)
+	u := nitridingSrv + pathRoot
+	// Make sure that the Internet-facing Web server is already running because
+	// we didn't ask nitriding to wait for the application.  The Web server may
+	// not be running by the time we test it, so we back off a few times, to
+	// give the Web server time to start.
+	func(t *testing.T, u string) {
+		for i := 0; i < 100; i += 10 {
+			resp, err := http.Get(u)
+			// The server probably isn't ready yet.  Sleep briefly.
+			if errors.Is(err, syscall.ECONNREFUSED) {
+				time.Sleep(time.Millisecond * time.Duration(i))
+				continue
+			}
+			if err != nil {
+				t.Fatalf("Expected no error but got %v", err)
+			}
+			if resp.StatusCode == http.StatusOK {
+				return
+			}
+		}
```

Comment:
Do you need a `t.Fatal("Timeout waiting for internet-facing server to become available")` here to detect startup failure?

Reply:
Addressed in ff190ab.